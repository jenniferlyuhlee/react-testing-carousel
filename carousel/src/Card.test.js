import { render, fireEvent } from "@testing-library/react";
import Card from "./Card";
import TEST_IMAGES from "./_testCommon.js";


// smoke test
it("renders without crashing", function() {
    render(<Card
            caption= "Test card"
            src = {TEST_IMAGES[0].src}
            currNum = {1}
            totalNum = {10} />);
});

// snapshot test
it ("matches snapshot", function () {
    const {asFragment} = render(<Card
                            caption= "Test card"
                            src = {TEST_IMAGES[0].src}
                            currNum = {1}
                            totalNum = {10} />);
    expect(asFragment()).toMatchSnapshot();
});