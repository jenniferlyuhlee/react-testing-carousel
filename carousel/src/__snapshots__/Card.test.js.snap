// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      Test card
    </h4>
    <img
      alt="Test card"
      class="Card-image"
      src="test1.com"
    />
    <small
      class="Card-small"
    >
      Image 1 of 10.
    </small>
  </div>
</DocumentFragment>
`;
